runtime/src/node_modules/@sapper/internal/shared.d.ts

```diff
@@ -0,0 +1,11 @@
+export const CONTEXT_KEY: unknown
+
+export interface ComponentModule {
+  default: ComponentConstructor
+  preload: (props: { params: Params, query: Query }) => Promise<any>;
+}
+
+export interface ComponentConstructor {
+  new(options: { target: unknown, props: unknown, hydrate: boolean }): Component;
```

Comment:
If props are the props of the Svelte component, and if you do the proposed change for `Component` in my other comment, then you could change this to

```ts
export interface ComponentConstructor<Props = any> {
  new(options: { target: unknown, props: Props, hydrate: boolean }): Component<Props>;
  preload: (props: { params: Params, query: Query }) => Promise<any>;
}
```

Reply:
I'm not quite sure I understand the suggestion to make this a class